flink-streaming-scala/src/main/scala/org/apache/flink/streaming/api/scala/WindowedStream.scala

```diff
@@ -362,38 +362,84 @@ class WindowedStream[T, K, W <: Window](javaStream: JavaWStream[T, K, W]) {
 
   /**
    * Applies an aggregation that that gives the maximum element of the window by
-   * the given position. When equality, returns the first.
+   * the given position. When equality, returns the first by default.
    */
   def maxBy(position: Int): DataStream[T] = aggregate(AggregationType.MAXBY,
-    position)
+    position, true)
 
   /**
    * Applies an aggregation that that gives the maximum element of the window by
-   * the given field. When equality, returns the first.
+   * the given position. When equality, returns either the first or last one depending
+   * on the parameter setting.
+   */
+  def maxBy(position: Int, first: Boolean): DataStream[T] = aggregate(AggregationType.MAXBY,
+    position, first)
+
+  /**
+   * Applies an aggregation that that gives the maximum element of the window by
+   * the given field. When equality, returns the first by default.
    */
   def maxBy(field: String): DataStream[T] = aggregate(AggregationType.MAXBY,
-    field)
+    field, true)
+
+  /**
+   * Applies an aggregation that that gives the maximum element of the window by
+   * the given field. When equality, returns either the first or last one depending
+   * on the parameter setting.
+   */
+  def maxBy(field: String, first: Boolean): DataStream[T] = aggregate(AggregationType.MAXBY,
+    field, first)
 
   /**
    * Applies an aggregation that that gives the minimum element of the window by
-   * the given position. When equality, returns the first.
+   * the given position. When equality, returns the first by default.
    */
   def minBy(position: Int): DataStream[T] = aggregate(AggregationType.MINBY,
-    position)
+    position, true)
 
   /**
    * Applies an aggregation that that gives the minimum element of the window by
-   * the given field. When equality, returns the first.
+   * the given position. When equality, returns either the first or last one depending
+   * on the parameter setting.
+   */
+  def minBy(position: Int, first: Boolean): DataStream[T] = aggregate(AggregationType.MINBY,
+    position, first)
+
+  /**
+   * Applies an aggregation that that gives the minimum element of the window by
+   * the given field. When equality, returns the first by default.
    */
   def minBy(field: String): DataStream[T] = aggregate(AggregationType.MINBY,
-    field)
+    field, true)
+
+  /**
+   * Applies an aggregation that that gives the minimum element of the window by
+   * the given field. When equality, returns either the first or last one depending
+   * on the parameter setting.
+   */
+  def minBy(field: String, first: Boolean): DataStream[T] = aggregate(AggregationType.MINBY,
+    field, first)
+
+  private def aggregate(aggregationType: AggregationType, position: Int): DataStream[T] = {
```

Comment:
This method may not be changed to `private` because it is a stable `@Public` interface.